runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -115,14 +118,24 @@ public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
                 .build();
 
         Optional<MeterRegistry> metricsRegistry = MeterRegistryProvider.getInstance();
-        MicroMeterUtils.gauge("address_space.read_cache.hit_ratio", readCache, cache -> cache.stats().hitRate());
         metricsRegistry.map(registry -> GuavaCacheMetrics.monitor(registry, readCache, "address_space.read_cache"));
+        MicroMeterUtils.gauge("address_space.read_cache.hit_ratio", readCache, cache -> cache.stats().hitRate());
+        MicroMeterUtils.gauge("address_space.read_cache.size", readCache, cache -> cache.size());
+        MicroMeterUtils.gauge("address_space.read_cache.avg_entry_size", calculateEstimatedAvgEntrySize());
```

Comment:
Ignoring how accurate the estimate function is, another way to improve this patch's accuracy is to keep two counters, one for the total number of entries and the other one for total of all estimates. To compute the average you just need to divide both counters and reset them to 0. That way you dont have to account for evictions.